contracts/src/StakeTableV2.sol

```diff
@@ -9,18 +9,38 @@ import { EdOnBN254 } from "./libraries/EdOnBn254.sol";
 import { BN254 } from "bn254/BN254.sol";
 import { BLSSig } from "./libraries/BLSSig.sol";
 
+/// @title Ethereum L1 component of the Espresso Global Confirmation Layer (GCL) stake table.
+///
+/// @dev All functions are marked as virtual so that future upgrades can override them.
+///
+/// @notice This contract is an upgrade to the original StakeTable contract. On Espresso mainnet we
+/// will only use the V2 contract. On decaf the V2 is used to upgrade the V1 that was first deployed
+/// with the original proof of stake release.
+///
+/// @notice The V2 contract contains the following changes:
+///
+/// 1. The functions to register validators and update consensus keys are updated to require both a
+/// BLS signature and a Schnorr signature and emit the signatures via events so that the GCL can
+/// verify them. The new functions and events have a V2 postfix. After the upgrade components that
+/// support registration and key updates must use the V2 functions and listen to the V2 events. The
+/// original functions revert with a `DeprecatedFunction` error in V2.
+///
+/// 2. The exit escrow period can be updated by the owner of the contract.
+///
+/// @notice The StakeTableV2 contract ABI is a superset of the original ABI. Consumers of the
+/// contract can use the V2 ABI, even if they would like to maintain backwards compatibility.
 contract StakeTableV2 is StakeTable {
     // === Events ===
 
     /// @notice A validator is registered in the stake table
     /// @notice the blsSig and schnorrSig are validated by the Espresso Network
     event ValidatorRegisteredV2(
         address indexed account,
-        BN254.G2Point blsVk,
-        EdOnBN254.EdOnBN254Point schnorrVk,
+        BN254.G2Point blsVK,
+        EdOnBN254.EdOnBN254Point schnorrVK,
         uint16 commission,
         BN254.G1Point blsSig,
```

Comment:
since we made schnorrSig bytes, any chance it makes sense to do so for blsSig too? cc @alxiong

Reply:
But IIRC we are missing some code so if we don't use bytes things would have gotten more complicated.

Reply:
personally I like `bytes` because it's more future proof. but whenever we want to do something with these objects in the contract, we should consider giving them a concrete types. 

In this case, we do verify the BLS signature, thus giving it a type is better IMO.

I hear Mathis's complain about no `Copy` for bytes, but my personal tolerance for `Clone` is higher and I can live with lots of `.clone()` everywhere, (clippy will inform us about unnecessary clone once alloy bindings decide to add Copy on them)